blog/_posts/2023-04-05-codespaces-teaching.md

```diff
@@ -0,0 +1,38 @@
+---
+layout: page
+title:  GitHub Codespaces for teaching
+date:   2023-04-05
+subheadline: "Training"
+teaser: "
+ GitHub Codespaces offering an interesting option for easy to spin up computational teaching environments
+"
+author: Alex Coleman
+image:
+  thumb: blog/codespaces-teaching/1200x630-codespaces-social.png
+
+breadcrumb: true
+---
+
+_Thumbnail taken from [GitHub blog post](https://github.blog/2021-08-11-githubs-engineering-team-moved-codespaces/). All copyright for the image belongs to GitHub._
+
+Recently, we delivered a Python and git/GitHub crash course session for a centre for doctoral training. It’s always great to do courses with a room full of keen PhD students and this was no exception. The catch with this course was the limited time frame required to both introduce Python and also git and GitHub. Our internal preference is to do this as two separate courses, a 2-day Python course and a 1-day intro to version control using git and GitHub. 
+
+A critical steep learning curve for both these courses is often set up. We encourage people to bring their own devices or use a University cluster machine and typically try and use tools like conda to help manage installation. This however adds additional complexity which feels like an extra cognitive burden for learners. Now they have to think about environments, directories to install things, etc etc. In recent courses we've accepted this and spent some time talking about these things in the time we’ve had available but for a 1-day course touching on both we wanted to try something a little different. For this course setup need to be easy so we didn’t have to think about installing Python or git, and be able to get straight into writing Python with a bit of git thrown in as well as integrating with GitHub.
```

Comment:
I have no idea why, but the apostrophe is being rendered differently through the text.
See the example with "we've" bellow

```
In recent courses we've accepted this and spent some time talking about these things in the time we’ve had...
```